rx/core/observer/observer.py

```diff
@@ -9,26 +9,45 @@ class enforces the grammar of observers where OnError and
     OnCompleted are terminal messages.
     """
 
+    @staticmethod
+    def default_on_next(value: Any) -> None:
+        """Default `on_next()` function. Does nothing."""
+        pass
+
+    @staticmethod
+    def default_on_error(error: Exception) -> None:
+        """Default `on_error()` function. Raises the given exception."""
+        if isinstance(error, BaseException):
+            raise error
+        else:
+            raise Exception(error)
+
+    @staticmethod
+    def default_on_completed() -> None:
+        """Default `on_completed()` function. Does nothing."""
+        pass
+
     def __init__(self,
                  on_next: Optional[typing.OnNext] = None,
                  on_error: Optional[typing.OnError] = None,
                  on_completed: Optional[typing.OnCompleted] = None
                  ) -> None:
+
         self.is_stopped = False
-        if on_next is not None:
-            self._on_next_core = on_next
-        if on_error is not None:
-            self._on_error_core = on_error
-        if on_completed is not None:
-            self._on_completed_core = on_completed
+        self._handler_on_next = on_next or Observer.default_on_next
+        self._handler_on_error = on_error or Observer.default_on_error
+        self._handler_on_completed = on_completed or Observer.default_on_completed
```

Comment:
Here you can use `noop` and `default_error` instead of the static methods. See `AutoDetachObserver`.

Reply:
thanks! I was looking for something like that but I missed these ones.